### PR TITLE
Validation ci update

### PR DESCRIPTION
**Pull request by FynnBe**
changes as discussed in https://github.com/bioimage-io/bioimage.io/issues/28#issuecomment-954768073

todo: 
 - [x] skip dynamic testing based on `run_mode`  (got a special error message in bioimageio.core)

### CODE REVIEW

**Comment by FynnBe** [comment]
preview of gh-pages here: https://github.com/bioimage-io/collection-bioimage-io/tree/gh-pages-validation_ci_update

**Comment by FynnBe** [comment]
examples:
-  static failure: https://github.com/bioimage-io/collection-bioimage-io/blob/gh-pages-validation_ci_update/summaries/10.5072/zenodo.849032.json
- dynamic failure: https://github.com/bioimage-io/collection-bioimage-io/blob/gh-pages-validation_ci_update/summaries/10.5072/zenodo.849110.json

**Review by constantinpape — APPROVED** [review]
LGTM, but I think the opinion of @oeway is more important here; as he has to parse the results to display badges on bioimage.io.

**Comment by oeway** [comment]
Hi @FynnBe It's nice that you now have static and dynamic error separated. Thinking about the structure, would it be better to organize the status/error into a list? Then the website rendering can just display it as a bullet points similar to github actions ([example](https://github.com/bioimage-io/collection-bioimage-io/runs/4078492797?check_suite_focus=true)) and we don't need to know what check happens first.

Here is what I would propose:
```json
{
"py3.9.7_pytorch_state_dict": [
 {"name": "static check", "success": true},
 {"name": "dynamic check", "success": true},
 {"name": "test model execution", "success": false, "error": "...", "traceback": "..."}
],
"py3.9.7_onnx": [
 {"name": "static check", "success": true},
 {"name": "dynamic check", "success": false, "error": "...", "traceback": "..."},
 {"name": "test model execution", "success": false, "error": "skipped due to failed dynamic check"}
]
}
```

The advantage is that we will be able to extend the check list with more steps and it's easy to know what happens first.

**Comment by FynnBe** [comment]
Sounds good. I'll adapt it accordingly 👍 

just as a side note to your example: "dynamic check" == "test model execution".

So, it'll be the static and the test model execution. Maybe we can call that step "reproduced test outputs" to make it explicit what this dynamic check is all about?

**Comment by FynnBe** [comment]
... and maybe "static resource format validation" instead of "static check" to be more informative?

**Comment by FynnBe** [comment]
@oeway you also omitted the environment name in your example.

I am aiming at this now:
```json
{
    "validate-bio-core-mix": {
        "py3.9.7_pytorch_state_dict": [
            {"name": "static resource format validation", "success": true},
            {"name": "reproduced test outputs", "success": true},
        ],
        "py3.9.7_onnx": [
            {"name": "static resource format validation", "success": false, "error": "...", "traceback": "..."},
            {"name": "reproduced test outputs", "success": false, "error": "skipped due to invalid resource format", "traceback": null}
        ]
    }
}

```

**Comment by oeway** [comment]
> @oeway you also omitted the environment name in your example.
> 
> I am aiming at this now:
> 
> ```json
> {
>     "bio-core-mix": {
>         "py3.9.7_pytorch_state_dict": [
>             {"name": "static resource format validation", "success": true},
>             {"name": "reproduced test outputs", "success": true},
>         ],
>         "py3.9.7_onnx": [
>             {"name": "static resource format validation", "success": false, "error": "...", "traceback": "..."},
>             {"name": "reproduced test outputs", "success": false, "error": "skipped due to invalid resource format", "traceback": null}
>         ]
>     }
> }
> ```

Sounds great! It was just a rough sketch, feel free to adapt it as you like!
